test/controllers/responses_controller_test.rb

```diff
@@ -1,7 +1,62 @@
 require 'test_helper'
 
 class ResponsesControllerTest < ActionDispatch::IntegrationTest
-  # test "the truth" do
-  #   assert true
-  # end
+  include Devise::Test::IntegrationHelpers
+
+  setup do
+    @user     = create(:user)
+    @admin    = create(:admin)
+    @survey   = create(:survey, user: @user)
+    @response = create(:response, survey: @survey)
+  end
+
+  test "response can only be destroyed by a logged in user" do
+    assert_difference('Response.count', 0) do
+      delete survey_response_url(@survey, @response)
+    end
+    assert_redirected_to new_user_session_path
+  end
+
+  test "response can only be deleted by owner of the survey" do
+    @survey2   = create(:survey, user: @admin)
+    @response2 = create(:response, survey: @survey2)
+
+    sign_in @user
+    delete survey_response_url(@survey2, @response2)
+    assert_redirected_to survey_path(@survey2)
+  end
+
+  test "a user cannot delete another user's responses" do
+    @user2    = create(:user)
+
+    sign_in @user2
+    delete survey_response_url(@survey, @response)
+    assert_redirected_to survey_path(@survey)
+  end
+
+  test "an admin can delete any response" do
+    sign_in @admin
+
+    assert_difference('Response.count', -1) do
+      delete survey_response_url(@survey, @response)
+    end
+  end
+
+  test "show can not be accessed without being logged in" do
+    get survey_response_url(@survey, @response)
+    assert_redirected_to new_user_session_path
+  end
+
+  test "new can be accessed by anyone" do
```

Comment:
Let's do a test to make sure this can only be accessed if the survey is marked as published. An unpublished survey should redirect you back to the home page

Reply:
Sure thing @spencerldixon